packages/sdk38/src/wallet.ts

```diff
@@ -0,0 +1,131 @@
+import {
+  Bip39,
+  EnglishMnemonic,
+  Secp256k1,
+  Sha256,
+  Sha512,
+  Slip10,
+  Slip10Curve,
+  Slip10RawIndex,
+} from "@cosmjs/crypto";
+
+import { rawSecp256k1PubkeyToAddress } from "./address";
+
+export type PrehashType = "sha256" | "sha512" | null;
+
+// TODO: Make this a union type
+export type Algo = "secp256k1";
+
+export interface AccountData {
+  // bech32-encoded
+  readonly address: string;
+  readonly algo: Algo;
+  readonly pubkey: Uint8Array;
+}
+
+export interface Wallet {
+  /**
+   * Request access to the user's accounts. Wallet should ask the user to approve or deny access. Returns true if granted access or false if denied.
+   */
+  readonly enable: () => Promise<boolean>;
```

Comment:
I wonder if a wallet can be disabled over time. In this case, the caller needs to call `.enable()` before every wallet interaction. In this case, the method gets pointless as the mechanism could and should be integrated in `getAccounts` and `sign`.

Reply:
🤔